Added dump from GE helper docs

getorymckeag opened the pull request:
I sanitized some help documentation that I thought would be helpful for TROLIE.  Adds some markdown pages to the Jekyll site, including:

- Overall discussion on the usage of REST.  This is not meant to be an ADR- rather, it's a narrative for people learning about TROLIE as to what it's not ICCP, or CSV files or something else they would come in biased towards.  
- Some concept definitions.  Should obviously be expanded upon
- Walkthrough examples with Curl.  FWIW, I tried to see if we can reuse JSON examples from the spec in these pages using Liquid templates.  The closest I came was to do that and run it through the jsonify function, but there isn't a way to pretty-print it.  
- Some explicit discussion of why RFC 3339 helps make DST less of a nightmare.

getorymckeag committed:
Added dump from GE helper docs

caindy commented:
This is great! I opened a draft PR against this branch to highlight three ways to reuse JSON examples without losing formatting. #30 

That PR is meant to make it easy to review; I don't want to merge it. Instead, once we pick a way, we'll just update this branch.

I'd also like an opportunity to proofread this and offer some suggested edits.

caindy commented on docs/rest.md:
```diff
@@ -0,0 +1,63 @@
+The Inter-Control Center Communications Protocol (ICCP), part of IEC standard 60870-6, is used freqently to integrate reliability coordinators' SCADA system with member transmission owners' SCADA systems to capture all kinds of telemetry data from the power grid.  It has traditionally been used for dynamic line ratings / real-time AARs for selected lines as an emerging technology.  ICCP works wel ...
+ICCP is reliable, mature, and well-known, and will continue to be supported by many transmission providers as a method for submitting real-time ratings.  Even though TROLIE supports real-time rating exchange, continued use of ICCP for real-time ratings is compatible with the use of TROLIE.  However, ICCP won’t scale well to handle the forecasted data.  There are a couple reasons for this:
+*	The data model is a significant mismatch.  ICCP is designed around SCADA concepts, such as points and quality flags.  It is difficult to create arbitrarily complex data structures, like a coordinated forecast, and link them together.  There are also needs to provide various strings into this data, such as override reasons.  
+* Using ICCP for forecasts would result in a significant increase in the count of SCADA points.  Across our customer base, GE sees increased pressure on SCADA systems due to other changes to the power grid.  The increased configuration overhead and load on potentially overtaxed SCADA systems makes ICCP less attractive.  
```
We could keep the main idea here and make it more general, i.e., shift the voice away from GE and to TROLIE. However, I'd like to consider dropping the "why not ICCP?" rationale from the TROLIE docs OR we could make some references to the OpenCIP concept documentation to bring in a broader perspective about the future of web technologies in the Operational Technology (vs IT) space.

caindy committed:
formatting, spelling

caindy commented on docs/rest.md:
I'm not opposed to a treatment of the use of web technologies in an OT solution, but I'd like to have a shared understanding of the intended audience, goals, etc.

To this point I have only considered the audience for this site to be implementors and consumers of the TROLIE spec, so I've started with a few assumptions; among those are:

* The relative merits of a REST(ful) API versus heritage OT are prior knowledge.
* They are at least somewhat familiar OpenAPI and web technologies.
* In general, their documentation needs are to be informed not persuaded.

Of course, I'm coming this from a perspective where the decision has been made for well over a year that the real-time exchange will utilize heritage OT, so my stakeholders will be sending their vendors / developers here to figure out how to use this web API.

I also think that some in that implementor / consumer audience wouldn't have an "heritage OT" bias but rather a bias toward using the most modern web technologies. In other words, their question is "Why REST and OpenAPI *versus gRPC or GraphQL*?" as opposed to "*versus SCADA/ICCP*".

To be clear, I don't disagree that an architecture decision record for "What technology standards will TROLIE use to specify the exchange?" (or a similar question) is needed. Moreover, I recognize that some of the alternatives, selection criteria, and rationale to be contemplated in that decision record are already present in this document; when we write the ADR, we should be intentional about extracting those details from this text.

May I suggest that we add a "Background" section to the site, adding this article with you as the named author, and inserting a lede at the top that identifies the intended audience, something like "For power industry veterans, SCADA or a file-based exchange may seem like obvious choices to implement a ratings exchange. In this article we discuss why TROLIE opted to utilize an approach leveraging well-established web technologies."

getorymckeag replied:
Yeah I agree, the way it will read will very much depend on the audience.  I do think this bias is here and needed; there is a very large subset of consumers and stakeholders of TROLIE that come from a heavy OT bias, which is obviously why I wrote this.  Last week, while we've been discussing this whole concept forever, I had to defend to one of our executives the need for this app to exist, and why it isn't just a big pile of more SCADA points.  Not to say they were hard to convince, I just had to have the argument in my pocket.  I had to have an almost identical discussion with one of our modeling team leads yesterday.  

And as TO engineers engage with TROLIE directly, they will all come to it from an OT background.  My intent here is very much NOT an ADR, but a tailored narrative.  I feel like ADRs are too formal for this audience- I don't even want to use the term "OT", because that is what IT people call OT people- OT people very rarely call themselves that.  

I'll think about a way to organize it that will hopefully be less of a trigger / point of confusion to the audience members you're discussing.

caindy committed:
formatting, spelling

caindy committed:
grammar

caindy commented on docs/example-narratives/submitting-forecasts.md:
```diff
@@ -0,0 +1,87 @@
+-X PATCH \
+-H "Content-Type: application/vnd.trolie.rating-forecast-proposal.v1+json" \
+-o output.json \
+"https://trolie.transmission-provider.com/rating-proposals/forecasts"
```
Let's use [example.com](https://en.wikipedia.org/wiki/Example.com) in the docs, so this would be `https://trolie.example.com/rating-proposals/forecasts`. Alternatively, we could use a consistent placeholder, like `https://<your TROLIE host>/rating-proposals/forecasts`.

We should probably change this to match https://github.com/trolie/spec/blob/1.0.0-wip/docs/_data/openapi-split.yaml#L93

getorymckeag replied:
Agreed, done along with a commit that pulls in use of relative examples.

getorymckeag committed:
Added reused JSON examples, use of example.com as example URL

caindy committed:
make examples accessible via include_relative

caindy committed:
simplify some string definitions

reusing `#/components/schemas/description` without a linter error from the owasp rules is more trouble than it is worth

ETag should be opaque anyway
